src/lib/kw_config_loader.sh

```diff
@@ -358,8 +358,11 @@ function parse_configuration()
   if [ ! -f "$config_path" ]; then
     return 22 # 22 means Invalid argument - EINVAL
   fi
+
+  # @see https://stackoverflow.com/questions/12916352/shell-script-read-missing-last-line
+  # for information about the need for the second check in the while loop.
   # shellcheck disable=SC2162
-  while read line; do
+  while read line || [ -n "$line" ]; do
```

Comment:
Also, add a unit test for this behavior (config file without newline). Thinking of something like `test_parse_configuration_file_without_newline_at_the_end()` (please, enhance this name 😬), in which you take a sample config file without newline, run `parse_configuration` on it, and check if the configuration at the end of the file was parsed.